frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/SelectedNodeInputOutputTab.tsx

```diff
@@ -1,13 +1,93 @@
+import { ExecutionDetailsPropertiesValueCode } from '~/pages/pipelines/global/experiments/executions/details/ExecutionDetailsPropertiesValue';
+import { InputDefinitionParameterType } from '~/concepts/pipelines/kfTypes';
+import { NoValue } from '~/components/NoValue';
 
 type SelectedNodeInputOutputTabProps = {
   task: PipelineTask;
+  execution: Execution.AsObject | undefined;
 };
 
-const SelectedNodeInputOutputTab: React.FC<SelectedNodeInputOutputTabProps> = ({ task }) => {
+const SelectedNodeInputOutputTab: React.FC<SelectedNodeInputOutputTabProps> = ({
+  task,
+  execution,
+}) => {
+  const getExecutionFieldsMap = React.useCallback(
+    (key: string) =>
+      execution?.customPropertiesMap.find(([customPropKey]) => customPropKey === key)?.[1]
+        .structValue?.fieldsMap || [],
+    [execution?.customPropertiesMap],
+  );
+
+  const getExecutionValueFromInputType = React.useCallback(
+    (executionValues: ProtoValue.AsObject | undefined, type: InputDefinitionParameterType) => {
+      const { numberValue, boolValue, stringValue, listValue, structValue, nullValue } =
+        executionValues || {};
+
+      switch (type) {
+        case InputDefinitionParameterType.DOUBLE:
+        case InputDefinitionParameterType.INTEGER:
+          return numberValue;
+        case InputDefinitionParameterType.BOOLEAN:
+          return boolValue ? 'True' : 'False';
+        case InputDefinitionParameterType.STRING:
+          try {
+            const jsonStringValue = JSON.parse(stringValue ?? '');
+
+            if (parseFloat(jsonStringValue)) {
+              throw stringValue;
+            }
+
+            return (
+              <ExecutionDetailsPropertiesValueCode
+                code={JSON.stringify(jsonStringValue, null, 2)}
+              />
+            );
+          } catch {
+            return stringValue || <NoValue />;
+          }
+          break;
+        case InputDefinitionParameterType.LIST:
+          return <ExecutionDetailsPropertiesValueCode code={JSON.stringify(listValue, null, 2)} />;
+        case InputDefinitionParameterType.STRUCT:
+          return (
+            <ExecutionDetailsPropertiesValueCode code={JSON.stringify(structValue, null, 2)} />
+          );
+        default:
+          return nullValue;
+      }
+    },
+    [],
+  );
+
+  const getParams = React.useCallback(
+    (
+      params: PipelineTaskParam[] | undefined,
+      executionParamMapList: [string, ProtoValue.AsObject][],
+    ) =>
+      params?.map((taskParam) => {
+        const { label, value, type } = taskParam;
+        const executionParamMap = executionParamMapList.find(
+          ([executionInputKey]) => taskParam.label === executionInputKey,
+        );
+        const { 1: executionParamValues } = executionParamMap || [];
+        const executionValue = getExecutionValueFromInputType(executionParamValues, type);
+
+        return {
+          label,
+          value: value || executionValue || type,
```

Comment:
why fall back to execution value. Can we make it the first one, then fall back to value?

Reply:
There was no reason. If that's a more correct approach, I'll update. I would expect if there was a "value" that it would equal the "executionValue" so I don't think the order matters, but I could be wrong.